control/grpc.py

```diff
@@ -110,6 +150,8 @@ def create_bdev(self, request, context=None):
         except Exception as ex:
             self.logger.error(f"create_bdev failed with: \n {ex}")
             if context:
+                if got_omap_lock:
+                    self.omap_state.unlock_omap()
```

Comment:
I recommend utilizing a context manager wrapper, see https://book.pythontips.com/en/latest/context_managers.html. With this approach, you can acquire the lock in the enter method and automatically release it when leaving the context in the exit method, similar to how it's handled with the gateway object. Don't need to worry about leaving the lock locked accidentally in one of the execution paths.